flock_drone/mechanics/distance.py

```diff
@@ -0,0 +1,92 @@
+from haversine import haversine
+import math
+
+
+
+# Distances are measured in miles/Kilometers.
+# Longitudes and latitudes are measured in degrees.
+# Earth is assumed to be perfectly spherical.
+
+# earth_radius = 3960.0 ## For miles
+earth_radius = 6371.0 ## For Kilometers
+degrees_to_radians = math.pi/180.0
+radians_to_degrees = 180.0/math.pi
+
+def change_in_latitude(distance):
+    """Given a distance north, return the change in latitude."""
+    return (distance/earth_radius)*radians_to_degrees
+
+def change_in_longitude(latitude, distance):
+    """Given a latitude and a distance west, return the change in longitude."""
+    # Find the radius of a circle around the earth at given latitude.
+    r = earth_radius*math.cos(latitude*degrees_to_radians)
+    return (distance/r)*radians_to_degrees
+
+
+# print(change_in_latitude(-1))
+# print(change_in_longitude(a[0], -1))
+
+def convert_direction_to_north_or_west(distance_moved, direction):
+    """Convert East and South direction to North and East."""
+    if direction in ["S","E"]:
+        if direction == "S":
+            distance_moved = distance_moved * -1
+            direction = "N"
+        elif direction == "E":
+            distance_moved = distance_moved * -1
+            direction = "W"
+
+    return distance_moved, direction
+
+def gen_new_coordinates_from_change_in_coordinates(old_coordinates, change_in_coordinates):
+    """Calculate new coordinates given coordinates(lat,lon) and change_in_coordinates(lat,lon)."""
+    return tuple(map(lambda x, y: x - y, old_coordinates, change_in_coordinates))

```

Comment:
Change must be added I think. If the old coordinates are (1,1) and the change is (-1,0) it should be at (0,1) yes? Your function would return (2,1).
The change is delta, in most math notations delta is added to the original value.